src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -404,82 +455,82 @@ protected override void PostStop()
-                .With<StartupFinished>(sf => Context.Become(Accepting))
-                .With<ShutdownAndFlush>(sf =>
+            Receive<ListensResult>(listens =>
+            {
+                _transportMapping = (from mapping in listens.Results
+                                     group mapping by mapping.Item1.Address
+                                           into g
+                                     select new { address = g.Key, transports = g.ToList() }).Select(x =>
+                                     {
+                                         if (x.transports.Count > 1)
+                                         {
+                                             throw new RemoteTransportException(
+                                                 string.Format("There are more than one transports listening on local address {0}",
+                                                     x.address));
+                                         }
+                                         return new KeyValuePair<Address, AkkaProtocolTransport>(x.address,
+                                             x.transports.Head().Item1.ProtocolTransport);
+                                     }).ToDictionary(x => x.Key, v => v.Value);
+
+                //Register a listener to each transport and collect mapping to addresses
+                var transportsAndAddresses = listens.Results.Select(x =>
                 {
-                    Sender.Tell(true);
-                    Context.Stop(Self);
-                });
+                    x.Item2.SetResult(new ActorAssociationEventListener(Self));
+                    return x.Item1;
+                }).ToList();
+
+                listens.AddressesPromise.SetResult(transportsAndAddresses);
+            });
+            Receive<ListensFailure>(failure => failure.AddressesPromise.SetException(failure.Cause));
+
+            // defer the inbound association until we can enter "Accepting" behavior
+
+            Receive<InboundAssociation>(
+                ia => Context.System.Scheduler.ScheduleTellOnce(TimeSpan.FromMilliseconds(10), Self, ia, Self));
+            Receive<ManagementCommand>(mc => Sender.Tell(new ManagementCommandAck(status: false)));
+            Receive<StartupFinished>(sf => Become(Accepting));
+            Receive<ShutdownAndFlush>(sf =>
+             {
+                 Sender.Tell(true);
+                 Context.Stop(Self);
+             });
         }
 
         /// <summary>
         /// Message-processing behavior when the <see cref="EndpointManager"/> is able to accept
         /// inbound association requests.
         /// </summary>
-        /// <param name="message">Messages from local system and the network.</param>
-        protected void Accepting(object message)
+        protected void Accepting()
         {
-            message.Match()
-                .With<ManagementCommand>(mc =>
+            Receive<ManagementCommand>(mc =>
```

Comment:
`Receive<ManagementCommand>` has not fundamentally changed, aside from implementing @JeffCyr's TPL recommendations. Applies the `ManagementCommand` to ALL of the transports and replies back a "all acked" / "at least one did not ack" boolean value back to the sender.